File: birthday/birthday.go
package birthdaychocolate

func birthday(ns []int32, targetSum int32, targetSize int32) int32 {
	var result int32 = 0

	for i := 0; i < len(ns); i++ {
		var sum int32 = 0
		for j := i; j < i+int(targetSize) && j < len(ns); j++ {
			sum += ns[j]
		}

		if sum == targetSum {
			result++
		}
	}

	return result
}

File: birthday/Birthday.java
import java.util.List;

public class Birthday {

  public static int birthday(List<Integer> ns, int targetSum, int targetSize) {
    int result = 0;

    for (int i = 0; i < ns.size(); i++) {
      int sum = 0;
      for (int j = i; j < i + targetSize && j < ns.size(); j++) {
        sum += ns.get(j);
      }

      if (sum == targetSum) {
        result++;
      }
    }

    return result;
  }

}
